src/PrettyPrinting.jl

```diff
@@ -1601,13 +1608,18 @@ function _write_line(io::IOCustom, str::AbstractString)
     io.lowercasefirst = false
   end
   io.printed += textwidth(firststr)
-  reststr = str[firstlen + 1:end]
+  for index in restiter
+   reststr *= str[index]
+  end
   it = Iterators.partition(1:textwidth(reststr), c - ind > 0 ? c - ind : c)
   for i in it
-    written += write(io.io, "\n")
-    written += write_indent(io)
-    written += write(io.io, reststr[i])
-    io.printed = textwidth(reststr[i])
+   written += write(io.io, "\n")
+   written += write_indent(io)
+   for j in Base.Iterators.drop(Base.Iterators.take(str,i[end]), i[begin]-1)
```

Comment:
probably would be helpful to insert a comment explaining what's going on here...

Reply:
I've added some comments, not sure how useful they are. (Right now, in my mind, it is obvious what is going on, and I am having difficulty coming up with an explanation....)